### PR TITLE
Custom Pagination Using Django REST Framework with Vue.

### LINKED ISSUE #4250: Custom Pagination  Using Django rest framework and Vue.
#### Proposed title of article
Understanding Custom Pagination Using Django rest framework and Vue.

#### Proposed article introduction
On most modern web applications and websites, one will need to split the app content for better user experience and readability. This makes the app users focus on small content at a time. To do this with ease, Django comes with a pre-built class: Paginator, which creates and manages paginated data of applications using the default pagination styles.
Sometimes, one may want to customize and use a modified format to paginate data. This article will enable the reader to understand how to customize and modify the default pagination styles to suit users' needs using Django REST APIs with Vue. The reader will also learn the various ways to paginate data; Class-based views and Function-Based views and the importance of web pagination.
#### Key takeaways 
The reader will be able to:
- Understand what pagination is and its importance.
- Various ways of paginating web apps using Django.
- Understand how to customize and modify pagination styles using Django REST API with Vue.

#### Article quality
The article will help one understand how to use a modified format of the pagination styles by customizing the default pagination styles using Django REST Framework with Vue.
#### References
N/A

### Templates to use as guides
N/A

### PR DESCRIPTION
**Pull request by dentonya**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4250

### CODE REVIEW

**Comment by dentonya** [comment]
Any updates @jumamiller

File: content/articles/custom-pagination-using-django-rest-framework/index.md [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+- Have prior knowledge of Python and Python Django concepts.
+- Have a good understanding of Python Django REST Framework and Vue.
+- Have a suitable IDE installed such as Pycharm, VS Code, etc.
+
+For this article will use the Pycharm IDE, use this [page](https://www.jetbrains.com/pycharm/download/#section=windows) to download and install it on your machine.
+
+### Pagination
+This is the process of splitting the content of a website or web application into discrete pages to make it more efficient to use and read when working with huge lists of data. Python Django comes with a pre-built class, **Paginator** for creating the paginations and managing the paginated data using the default styles. The process can be done either with **Class-Based Views** or **Function-Based Views** as will be discussed in the tutorial.
+Sometimes, one may want to customize and modified the pagination format, this is made possible using **Django REST Framework** which includes support for customized pagination styles. This pagination API may handle links that are embedded in the response's text or URLs that are contained in response headers like `Content-Range `or `Link`.
```
Comment (jumamiller) [inline content/articles/custom-pagination-using-django-rest-framework/index.md:42]: Hi Denno, this paragraph here is quite stacked, could you break it down..

File: content/articles/custom-pagination-using-django-rest-framework/index.md [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+            </div>
+            {% endfor %}
+        </div>
+        {% block sidebar %} {% include 'sidebar.html' %} {% endblock sidebar %}
+    </div>
+</div>
+{%endblock%}
+```
+Re-running the project the output below will be produced for class-based view pagination.
+![Class Based Views](/engineering-education/custom-pagination-using-django-rest-framework/class_based.jpg)
```
Comment (jumamiller) [inline content/articles/custom-pagination-using-django-rest-framework/index.md:173]: Explain the code snippets above, tell us walk us through the template!

File: content/articles/custom-pagination-using-django-rest-framework/index.md [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+                    }
+                }
+            }
+
+            Vue.createApp(Blog).mount('#blog')
+        </script>
+    </body>
+</html>
+```
+Re-running the program the following output should be displayed containing the next and previous buttons.
```
Comment (jumamiller) [inline content/articles/custom-pagination-using-django-rest-framework/index.md:425]: Explain this snippet too!

**Review by jumamiller — DISMISSED** [review]
@dentonya , I've requested for a few changes,kindly work on them before we move to tech review!

File: content/articles/custom-pagination-using-django-rest-framework/index.md (hunk 3 above)
Comment (dentonya) [reply content/articles/custom-pagination-using-django-rest-framework/index.md:425]: ok noted the changes @jumamiller

**Comment by dentonya** [comment]
> @dentonya , I've requested for a few changes,kindly work on them before we move to tech review!

hey @jumamiller done with changes

**Comment by jumamiller** [comment]
Hi @dentonya, could you check the hero image...upload news one.

**Comment by dentonya** [comment]
> Hi @dentonya, could you check the hero image...upload news one.

ok noted @jumamiller

**Comment by dentonya** [comment]
> Hi @dentonya, could you check the hero image...upload news one.

done with changes @jumamiller

**Comment by hectorkambow** [comment]
@dentonya 

Please be sure to go over any edits made by your peer reviewer to improve future articles and reviews and prevent any PR closure.

As a rule of thumb - we can only accept articles (pull requests) that are fully polished and complete articles - articles that are ready to be published.
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue).

**Comment by hectorkambow** [comment]
@jumamiller 

Can we get a status update on this PR? 

Lets also please refrain from making major edits/revision to the PRs - we can guide and point out certain issue for the author - but it is the author's responsibility to go through the content for any other issues or errors.

**Comment by LinusMuema** [comment]
Awesome job on the article @dentonya 

cc: @jumamiller

**Pull request merged**